packages/react/src/components/UIShell/HeaderPanel.js

```diff
@@ -34,6 +37,15 @@ const HeaderPanel = React.forwardRef(function HeaderPanel(
     [customClassName]: !!customClassName,
   });
 
+  if (__DEV__) {
```

Comment:
I think in the recent refactor of `warning()` they included this condition on `__DEV__` internally so we can unconditionally call `warning()` in the components.